diff --git a/config.d/99_custom_config b/config.d/99_custom_config
index 8b21924..48ea0f8 100644
--- a/config.d/99_custom_config
+++ b/config.d/99_custom_config
@@ -21,4 +21,8 @@ NUM_SLOTS = 1
 NUM_SLOTS_TYPE_1 = 1
 SLOT_TYPE_1 = 100%
 SLOT_TYPE_1_PARTITIONABLE = TRUE
-STARTD_ATTRS = $(STARTD_ATTRS) SiteName
\ No newline at end of file
+STARTD_ATTRS = $(STARTD_ATTRS) SiteName
+SINGULARITY_JOB = !isUndefined(TARGET.SingularityImage)
+SINGULARITY_IMAGE_EXPR = TARGET.SingularityImage
+SINGULARITY_BIND_EXPR = "/cvmfs"
+SINGULARITY_RUN_TEST_BEFORE_JOB = false
\ No newline at end of file
diff --git a/default-local/images.dodas.infn.it.conf b/default-local/images.dodas.infn.it.conf
new file mode 100644
index 0000000..5db9f78
--- /dev/null
+++ b/default-local/images.dodas.infn.it.conf
@@ -0,0 +1,7 @@
+CVMFS_SERVER_URL=http://131.154.97.118/cvmfs/images.dodas.infn.it
+CVMFS_PUBLIC_KEY=/etc/cvmfs/keys/images.dodas.infn.it/images.dodas.infn.it.pub
+CVMFS_HTTP_PROXY=DIRECT
+CVMFS_DEBUGFILE=/tmp/cvmfs_images_dodas.log
+CVMFS_LOGFILE=/tmp/cvmfs_images_dodas.log
+CVMFS_CACHE_BASE=/opt/cache
+CVMFS_QUOTA_LIMIT=4000
\ No newline at end of file
diff --git a/docker-compose.yaml b/docker-compose.yaml
index 977ef33..8b4479e 100644
--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -3,7 +3,7 @@ services:
   wn:
     depends_on:
     - cvmfs
-    image: ghcr.io/dodas-ts/htc-dask-wn:v0.0.26-dask 
+    image: ghcr.io/comp-dev-cms-ita/htcondor-dask-wn:v1.1.2
     restart: always
     privileged: true
     network_mode: "host"
@@ -42,7 +42,7 @@ services:
     restart: always
     #entrypoint: ["sleep", "infinity"]
     environment:
-    - REPO_LIST=cms.cern.ch  grid.cern.ch cms.dodas.infn.it 
+    - REPO_LIST=cms.cern.ch  grid.cern.ch cms.dodas.infn.it images.dodas.infn.it unpacked.cern.ch 
     privileged: true
     network_mode: "host"
     healthcheck:
@@ -63,6 +63,12 @@ services:
     - type: bind
       source: ./cms-dodas-key
       target: /etc/cvmfs/keys/cms.dodas.infn.it
+    - type: bind
+      source: ./images-dodas-key
+      target: /etc/cvmfs/keys/images.dodas.infn.it
+    - type: bind
+      source: ./cvmfs/cache
+      target: /opt/cache/
 
   # docker run --user telegraf:$(stat -c '%g' /var/run/docker.sock) -v /var/run/docker.sock:/var/run/docker.sock -v $PWD/telegraf-config/telegraf.conf:/etc/telegraf/telegraf.conf:ro telegraf  
   # GID=$(stat -c '%g' /var/run/docker.sock)  docker-compose up telegraf
diff --git a/images-dodas-key/images.dodas.infn.it.pub b/images-dodas-key/images.dodas.infn.it.pub
new file mode 100644
index 0000000..36673b6
--- /dev/null
+++ b/images-dodas-key/images.dodas.infn.it.pub
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAs0nzNysZ7WnG4uXaugi/
+gt3fzYc/gz3mhdKqbmHNKDVbD7afKsUdTzH4J97GQ04KLLCMx325DPkO3SdloQ+c
+PCfzKTz3Mu9aOiHtnmaWK1lk5K80e6QVDe/JMsPISGDKMeyDzQyRaZuCOne8sTLt
+lTkerTg0m90m8mRfHXzPQlaEFbifSclCfDmbhzgRbmGkdbKGosHBC47ZphpB/0x6
+BmDPhwQOlsPKgIUYlWt0gbITkYsUWEiajesxKFCOzzODSwvxVrYwsFK5og2Y7RBh
+q3//UuHvFKm0OfYG20cL8bFFAQDzXbgWv3qjbqaFMcH/JPtHY15iIbbsS26pso/P
+/QIDAQAB
+-----END PUBLIC KEY-----
\ No newline at end of file
